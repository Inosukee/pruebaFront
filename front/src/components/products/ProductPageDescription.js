import React, { useEffect, useState } from 'react'
import { getProductDesc } from '../../helpers/getProduct'

import PropTypes from 'prop-types'

export const ProductPageDescription = ({ id }) => {

  const [productDesc, setProductDesc] = useState({})


  useEffect(() => {
    getProductDesc(id).then((res) => {
      setProductDesc(res)
    })
  }, [id])

  return (
    <>
      <div className="product__desc">
        <h2>
          Descripci√≥n del producto
        </h2>
        <p className="product__desc__text">{productDesc.plain_text}</p>
      </div>
    </>
  )
}


ProductPageDescription.propTypes = {
  id: PropTypes.string.isRequired
}