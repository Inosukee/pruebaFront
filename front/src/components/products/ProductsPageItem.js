import React from 'react'
import CurrencyFormat from 'react-currency-format';
import { Link } from 'react-router-dom';

export const ProductsPageItem = ({ id, thumbnail, price, title, last = false }) => {
  return (
    <>
      <Link to={`./items/${id}`} className='product__grid' >
        <div className='product__container__img'>
          <img className='product__img' src={thumbnail} alt="foto" />
        </div>
        <div className='product__container__info'>
          <div className='product__container__price'>
            <CurrencyFormat value={price} displayType={'text'} thousandSeparator={true} prefix={'$ '} />
          </div>
          <div className='product__container__title'>
            {title}
          </div>
        </div>
      </Link>
      {
        !last ?
          (
            <hr />
          )
          :
          (
            <br />
          )
      }

    </>
  )
}
