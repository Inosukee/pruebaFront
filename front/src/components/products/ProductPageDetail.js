import React, { useEffect, useState } from 'react'
import { useParams } from 'react-router-dom';
import { getProductByid } from '../../helpers/getProduct';
import { ProductPageDescription } from './ProductPageDescription';
import CurrencyFormat from 'react-currency-format';

export const ProductPageDetail = () => {
  const [product__detail, setproduct__detail] = useState({})

  let { id } = useParams();

  useEffect(() => {
    getProductByid(id).then((res) => {
      setproduct__detail(res)
    })
  }, [id])

  const { pictures } = product__detail;

  return (
    <>
      {(product__detail && pictures) &&
        <div className='main__main animate__animated animate__fadeIn'>
          <div className="product__main">
            <div className="product__detail">
              <div className='product__detail__img'>
                <img className="product__detail__img" src={pictures[0].url}
                  alt="" />
              </div>
              <div className="product__detail__info">
                <p>{product__detail.sold_quantity} vendidos</p>
                <h4>
                  {product__detail.title}
                </h4>
                <CurrencyFormat className="product__detail__price" value={product__detail.base_price} displayType={'text'} thousandSeparator={true} prefix={'$ '} />
                <div>
                  <button className="btn btn-primary">Comprar</button>
                </div>
              </div>
            </div>
            <ProductPageDescription id={id} />
          </div>
        </div>
      }
    </>
  )
}
