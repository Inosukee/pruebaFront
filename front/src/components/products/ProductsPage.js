import React from 'react'
import { ProductsPageItem } from './ProductsPageItem'
import PropTypes from 'prop-types'

export const ProductsPage = ({ productos }) => {
    return (
        <div className='product__main'>
            <div className='product__wrapper'>
                {
                    productos &&
                    productos.map((products, i, { length }) => {
                        if (i < length - 1) {
                            return (<ProductsPageItem
                                key={products.id}
                                {...products}
                            />)
                        }
                        else {
                            return (<ProductsPageItem last={true}
                                key={products.id}
                                {...products}
                            />)
                        }
                    })
                }
            </div>
        </div>
    )
}

ProductsPage.propTypes = {
    productos: PropTypes.array
}