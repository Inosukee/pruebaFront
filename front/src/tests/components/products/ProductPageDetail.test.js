import React from 'react'
import { mount } from 'enzyme';
import { ProductPageDetail } from '../../../components/products/ProductPageDetail';
import { MemoryRouter, Route } from 'react-router-dom';



describe('Pruebas en <ProductPageDetail />', () => {
  test('debe mostrar el componente correctamente', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/items/MLA912447790']}>
        <Route path='items/:id' component={ProductPageDetail} />
      </MemoryRouter>
    )
    expect(wrapper).toMatchSnapshot()
  })
})