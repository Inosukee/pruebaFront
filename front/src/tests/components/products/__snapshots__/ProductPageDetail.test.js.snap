// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <ProductPageDetail /> debe mostrar el componente correctamente 1`] = `""`;
